product/views.py

```diff
@@ -0,0 +1,187 @@
+                product_count = Product.objects.filter(subcategory__in=subcategories).count()
+                num_pages     = product_count // item_per_page if product_count % item_per_page == 0 else product_count // item_per_page + 1;
+                product_list  = Product.objects.filter(subcategory__in=subcategories).order_by(order_by)[offset:offset+item_per_page]
+            else:
+                product_count = Product.objects.all().count()
+                num_pages     = product_count // item_per_page if product_count % item_per_page == 0 else product_count // item_per_page + 1;
+                product_list  = Product.objects.all().order_by(order_by)[offset:offset+item_per_page]
+
+            data = []
+
+            for product in product_list:
+                data.append(product.get_info())
+
+            return JsonResponse({ "numPages" : num_pages, "data" : data}, status=200)
+
+        except ValueError:
+            return JsonResponse({ "message" : "VALUE_ERROR"}, status=400)
+        except TypeError:
+            return JsonResponse({ "message" : "TYPE_ERROR"}, status=400)
+        except FieldError:
+            return JsonResponse({ "message" : "FIELD_ERROR"}, status=400)
+
+
+class Search(View):
+    def get(self,request):
+        try:
+            search_word = request.GET.get('search_word')
+
+            if len(search_word) == 0 :
+                return JsonResponse({"data" : []}, status=200)
+
+            product_list = Product.objects.filter(name__contains=search_word)
+
+            data = []
+
+            for product in product_list:
+                data.append(product.get_info())
+
+            return JsonResponse({"data" : data}, status=200)
+
+        except ValueError:
+            return JsonResponse({ "message" : "VALUE_ERROR"}, status=400)
+        except TypeError:
+            return JsonResponse({ "message" : "TYPE_ERROR"}, status=400)
+        except FieldError:
+            return JsonResponse({ "message" : "FIELD_ERROR"}, status=400)
+
+class Detail(View):
+    def get(self, request, product_id):
+        try:
+            product = Product.objects.get(id=product_id)
+
+            bodyColor = []
+            inkColor  = []
+            options   = []
+
+            for product_option in ProductOption.objects.filter(product=product):
+                body         = {}
+                body['name'] = product_option.body_color.name
+                body['hex']  = product_option.body_color.hex_code
+
+                ink         = {}
+                ink['name'] = product_option.ink_color.name
+                ink['hex']  = product_option.ink_color.hex_code
+
+                option = {}
+                option['name'] = body['name'] + "/" + ink['name'] + "(" + product_option.thickness.value + "mm)" + "(재고" + str(product_option.stock) + "개)"
+                option['price'] = product.price + product_option.plus_price
```

Comment:
```suggestion
                body         = {
                    'name' : product_option.body_color.name
                    ....
                }
                
```
과 같이 선언하시면 불필요한 빈 딕셔너리를 선언하지 않아도 되고, 그 이름을 계속 쓰지 않아도 되어서 편리합니다 :)

Reply:
수정했습니다. 훨씬 깔끔해진 것 같아요 :)